src/black/comments.py

```diff
@@ -321,6 +321,117 @@ def generate_ignored_nodes(
             container = container.next_sibling
 
 
+def _find_compound_statement_context(
+    parent: Optional[LN],
+) -> tuple[Optional[Node], Optional[Node]]:
+    """Find compound statement and suite nodes for fmt: skip handling."""
+    if parent is None or parent.type != syms.simple_stmt:
+        return None, None
+
+    # Case 1: simple_stmt -> suite -> compound_stmt (after reformatting)
+    if (
+        parent.parent is not None
+        and parent.parent.type == syms.suite
+        and parent.parent.parent is not None
+    ):
+        assert isinstance(parent.parent, Node)
+        suite_node = parent.parent
+        assert isinstance(suite_node.parent, Node)
+        compound_parent = suite_node.parent
+        return compound_parent, suite_node
+
+    # Case 2: simple_stmt -> compound_stmt (original structure)
+    compound_types = (
+        syms.if_stmt,
+        syms.while_stmt,
+        syms.for_stmt,
+        syms.try_stmt,
+        syms.with_stmt,
+        syms.funcdef,
+        syms.classdef,
+    )
+    if parent.parent is not None and parent.parent.type in compound_types:
+        assert isinstance(parent.parent, Node)
+        compound_parent = parent.parent
+        # In original single-line structure, the simple_stmt IS the body
+        assert isinstance(parent, Node)  # simple_stmt is always a Node
+        return compound_parent, parent
+
+    return None, None
+
+
+def _get_compound_statement_header(
+    compound_parent: Node, suite_node: Node, parent: LN
+) -> list[LN]:
+    """Get header nodes for compound statement if it should be preserved."""
+    compound_types = (
+        syms.if_stmt,
+        syms.while_stmt,
+        syms.for_stmt,
+        syms.try_stmt,
+        syms.with_stmt,
+        syms.funcdef,
+        syms.classdef,
+    )
```

Comment:
Could `STATEMENT` from `nodes.py` be used? (Both here and above) It includes a few types that aren't included here, though. I'm not sure if that was an oversight, if they just aren't applicable here, or if it was intentional and they would cause issues.

If you do need this separate type, could it be brought to the top level (and possibly moving it to nodes.py itself) to avoid duplication?